app/src/main/java/com/github/warnastrophy/core/model/GpsService.kt

```diff
@@ -0,0 +1,225 @@
+package com.github.warnastrophy.core.model
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Looper
+import android.util.Log
+import com.github.warnastrophy.core.util.AppConfig
+import com.google.android.gms.location.*
+import com.google.android.gms.maps.model.LatLng
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+
+/**
+ * Service managing the user's GPS position. Maintains an observable state representing the current
+ * position and possible errors.
+ *
+ * @param applicationContext Application context to access location services.
+ */
+interface PositionService {
+  /** Observable state representing the current GPS position and metadata. */
+  val positionState: StateFlow<GpsPositionState>
+}
+
+val TAG = "GpsService"
+
+/**
+ * Implementation of the PositionService interface, handling GPS position updates and state.
+ *
+ * @property applicationContext Application context for accessing location services.
+ */
+class GpsService(
+    private val applicationContext: Context,
+) : PositionService {
+
+  /** Coroutine scope for background operations. */
+  private val serviceScope = CoroutineScope(Dispatchers.IO)
+
```

Comment:
We add a `LocationCallback` here as a private attribute so it can be used in `startLocationUpdates()`. The detailed explanation is in comment below.
```suggestion
private val locationCallback: LocationCallback = object : LocationCallback() {
        override fun onLocationResult(result: LocationResult) {
            val pos = result.lastLocation
            Log.d(TAG, "Location update received: $pos")
            if (pos == null) {
                setError("No location fix available")
            } else {
                updatePosition(LatLng(pos.latitude, pos.longitude))
                setSuccess("Position updated automatically")
            }
            // Note: We don't set setLoading(false) here, as updates are continuous.
        }

        override fun onLocationAvailability(availability: LocationAvailability) {
            if (!availability.isLocationAvailable) {
                setError("Location temporarily unavailable")
            } else {
                setSuccess("Location service is available")
            }
        }
    }
```